src/main/java/com/kodilla/ecommercee/service/ProductService.java

```diff
@@ -0,0 +1,51 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.domain.dto.ProductDto;
+import com.kodilla.ecommercee.exception.GroupNotFoundException;
+import com.kodilla.ecommercee.exception.ProductNotFoundException;
+import com.kodilla.ecommercee.repository.GroupRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class ProductService {
+    private ProductRepository productRepository;
+    private GroupRepository groupRepository;
+
+    @Autowired
+    public ProductService(ProductRepository productRepository, GroupRepository groupRepository) {
+        this.productRepository = productRepository;
+        this.groupRepository = groupRepository;
+    }
+
+    public List<Product> getProductList() {
+        return productRepository.findAll();
+    }
+
+    public Optional<Product> getProductById(long id) {
+        return productRepository.findById(id);
+    }
+
+    public void addProduct(Product product) throws GroupNotFoundException {
+        productRepository.save(product);
+    }
+
+    public void deleteProduct(long id) {
+        productRepository.deleteById(id);
+    }
+
+    public Product  updateProduct(ProductDto updatedProduct, long id) throws GroupNotFoundException {
+        Product product = productRepository.findById(id).orElseThrow(ProductNotFoundException::new);
+        if (updatedProduct.getName() != null) product.setName(updatedProduct.getName());
+        if (updatedProduct.getDescription() != null) product.setDescription(updatedProduct.getDescription());
+        if (updatedProduct.getPrice() != null) product.setPrice(updatedProduct.getPrice());
+        if (updatedProduct.getGroupId() != null) product.setGroup(groupRepository.findById(updatedProduct.getGroupId()).orElseThrow(GroupNotFoundException::new));
```

Comment:
Jeśli wszystkie atrybuty produktu muszą posiadać wartość to w klasie `product` przy geterach powinna znaleźć się adnotacja `@NotNull`.

Reply:
@radion358 autor uzywa metody http patch, nie put. Czyli przekazane dto, moze miec np tylko description zadeklarowane i pozwala to na podmiane jednego pola, nie calego obiektu

Reply:
@MARudnicki faktycznie mój błąd